tools/proto_tool/ProtoLoader.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2018 The StartupOS Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.startupos.tools.proto_tool;
+
+import com.google.protobuf.Message;
+import com.google.protobuf.TextFormat;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.lang.reflect.Method;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.tools.JavaCompiler;
+import javax.tools.JavaCompiler.CompilationTask;
+import javax.tools.StandardJavaFileManager;
+import javax.tools.ToolProvider;
+
+public class ProtoLoader {
+  public static void toProtobin(
+      List<Path> protos, Path protoTxt, Path protoBinOutput, String protoClassName)
+      throws Exception {
+
+    final Path tempDirectory = Files.createTempDirectory("protoLoader");
+
+    ArrayList<String> args = new ArrayList<>(protos.size() + 3);
+    args.add("/usr/bin/env");
+    args.add("bash");
+    args.add(Paths.get("./tools/protoc.sh").toAbsolutePath().toString());
+
+    for (Path proto : protos) {
+      Path resutingFile =
```

Comment:
`resutingFile->resultingFile`

Reply:
@oferb fixed in https://github.com/google/startup-os/pull/574/commits/0f4f6d6d49274a7362ee1d0270ec3c60ce3a7d39